### ISSUE TITLE
Incorrect output for tests/autodiff/reverse-while-loop.slang on Vulkan via glsl

### ISSUE THREAD

**Issue by expipiplus1** [shader-slang/slang#3017]
With the latest slang, this test passes for CPU, CUDA and DX12 (via dxvk), however Vulkan generates an incorrect value.

This isn't caught by CI, so perhaps it's some driver (530.41.03) bug on my machine? Examining the generated GLSL looks fine at least...

```
$ ./bin/linux-x64/debug/slang-test tests/autodiff/reverse-while-loop.slang
Supported backends: dxc glslang gcc genericcpp nvrtc llvm
Check gl,ogl,opengl: Not Supported


Check vk,vulkan: Supported
Check dx12,d3d12: Supported
Check dx11,d3d11: Supported
Check cuda: Supported
ignored test: 'tests/autodiff/reverse-while-loop.slang (dx11)'
passed test: 'tests/autodiff/reverse-while-loop.slang.1 (dx12)'
error: ERROR:
EXPECTED{{{
type: float
8.000000
0.013107
0.000000
0.000000
0.000000
}}}
ACTUAL{{{
type: float
8.000000
0.000000
0.000000
0.000000
0.000000
}}}

FAILED test: 'tests/autodiff/reverse-while-loop.slang.2 (vk)'
passed test: 'tests/autodiff/reverse-while-loop.slang.3 (cpu)'
passed test: 'tests/autodiff/reverse-while-loop.slang.4 syn (cuda)'

===
75% of tests passed (3/4), 1 tests ignored
===

failing tests:
---
tests/autodiff/reverse-while-loop.slang.2 (vk)
---
```

**Comment by csyonghe** [comment]
I think it is likely a driver bug. I tested it on a newer driver and it seemed to be correct again.

**Comment by csyonghe** [comment]
Swiftshader also produces the correct result on this test.

**Comment by expipiplus1** [comment]
Looks like a regression between `525.116.04` and `530.41.03`, however it's fixed on `535.86.05` ¯\_(ツ)_/¯
